version: "3.9"

services:

  db:
    image: postgres
    restart: on-failure
    environment:
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_DB=${POSTGRES_DB}
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - pg_data:/var/lib/postgresql/data/
    expose:
      - ${POSTGRES_PORT}
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $POSTGRES_USER" ]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    tty: true
    command: bash -c 'python manage.py migrate && python manage.py runserver 0.0.0.0:8080'
    ports:
      - '8080:8080'
    env_file:
      - .env
    volumes:
      - .:/config
    depends_on:
      db:
        condition: service_healthy

volumes:
  pg_data:

#    done




